src/main/java/de/tum/in/www1/artemis/web/rest/open/PyrisStatusUpdateResource.java

```diff
@@ -0,0 +1,63 @@
+package de.tum.in.www1.artemis.web.rest.open;
+
+import jakarta.servlet.http.HttpServletRequest;
+
+import org.springframework.context.annotation.Profile;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import de.tum.in.www1.artemis.security.annotations.EnforceNothing;
+import de.tum.in.www1.artemis.service.connectors.pyris.PyrisJobService;
+import de.tum.in.www1.artemis.service.connectors.pyris.PyrisStatusUpdateService;
+import de.tum.in.www1.artemis.service.connectors.pyris.dto.tutorChat.PyrisTutorChatStatusUpdateDTO;
+import de.tum.in.www1.artemis.service.connectors.pyris.job.TutorChatJob;
+import de.tum.in.www1.artemis.web.rest.errors.AccessForbiddenException;
+import de.tum.in.www1.artemis.web.rest.errors.ConflictException;
+
+/**
+ * REST controller for providing Pyris access to Artemis internal data
+ */
+@RestController
+@Profile("iris")
+@RequestMapping("api/public/pyris/pipelines/")
+public class PyrisStatusUpdateResource {
+
+    private final PyrisJobService pyrisJobService;
+
+    private final PyrisStatusUpdateService pyrisStatusUpdateService;
+
+    public PyrisStatusUpdateResource(PyrisJobService pyrisJobService, PyrisStatusUpdateService pyrisStatusUpdateService) {
+        this.pyrisJobService = pyrisJobService;
+        this.pyrisStatusUpdateService = pyrisStatusUpdateService;
+    }
+
+    /**
+     * {@code POST /api/public/pyris/pipelines/tutor-chat/runs/{runId}/status} : Set the status of a tutor chat job.
+     *
+     * @param runId           the ID of the job
+     * @param statusUpdateDTO the status update
+     * @param request         the HTTP request
+     * @throws ConflictException        if the run ID in the URL does not match the run ID in the request body
+     * @throws AccessForbiddenException if the token is invalid
+     * @return a {@link ResponseEntity} with status {@code 200 (OK)}
+     */
+    @PostMapping("tutor-chat/runs/{runId}/status")
+    @EnforceNothing // We do token based authentication
+    public ResponseEntity<Void> setStatusOfJob(@PathVariable String runId, @RequestBody PyrisTutorChatStatusUpdateDTO statusUpdateDTO, HttpServletRequest request) {
+        var job = pyrisJobService.getJobFromHeader(request);
```

Comment:
see my comment in PyrisJobService, it's unclear that authentication is verified here. Make it clear from the method name